client.go

```diff
@@ -576,7 +576,7 @@ func (sc *SnowthClient) ListActiveNodes() []*SnowthNode {
 
 // do sends a request to IRONdb.
 func (sc *SnowthClient) do(ctx context.Context, node *SnowthNode,
-	method, url string, body io.Reader) (io.Reader, http.Header, error) {
+	method, url string, body io.Reader, headers map[string]string) (io.Reader, http.Header, error) {
```

Comment:
What if you used http.Header instead of map[string]string as the type to pass in request header data? That would match with the type used to return response headers. http.Header is a map[string][]string internally and can be used with the range operator.

Reply:
`http.Header` is technically a multi-map, but this is still the better way to go since it uses the Go standard library type:
```go
type Header map[string][]string
```
Fixed.